File: php/09proc/proc40.php
<?php
function Exp1($x, $e){
        $dar = 1;
		$sum = 1;
        $fact = 1;
		$i = 1;
        while($dar / $fact > $e){
                $dar *= $x;
                $fact *= $i;
                $sum += $dar / $fact;
                $i++;
        }
        return $sum;
}
$hisob = 1;
$x = mt_rand(-10, 10);
echo "x = ".$x;
$e = array(0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9);
while($hisob <= 6){
	$index = mt_rand(0, 8);
	$nat = Exp1($x, $e[$index]);
	$nat = number_format($nat, 2);
	echo "<br>e = ".$e[$index];
	echo "<br>Exp1(".$x.",".$e[$index].") = ".$nat."<br><br>";
	$hisob++;
}
?>

File: cpp/09proc/Proc40.cpp
#include <iostream>
using namespace std;

double Exp1(double x, double e);

int main()
{
	//Task("Proc40");
	double x;
	cin >> x;
	
	double e;
	for (int i = 1; i <= 6; i++)
	{
		cin >> e;
		cout << Exp1(x, e);
	}
	
	return 0;
}

double Exp1(double x, double e)
{
	double jsh = 0, sum = 0;
	double deg = 1, fact = 1;
	jsh = deg / fact;
	int i = 1;
	while (jsh > e)
	{
		sum += jsh;
		deg *= x;
		fact *= i++;
		jsh = deg / fact;
	}
	
	return sum;
}